### PR TITLE
go/marshal: allow any struct names while unmarshaling.

### PR DESCRIPTION
**Pull request by vinibaggio**
I need to unmarshal Noms Structs into arbitrarily-named Golang structs. Since this is already planned (https://github.com/attic-labs/noms/issues/2376), I went ahead and removed the code pointed by @aboodman and added a test case for that scenario.

### CODE REVIEW

File: go/marshal/decode_test.go [hunk 1]
```diff
@@ -192,6 +192,15 @@ func TestDecode(tt *testing.T) {
 	t(types.NewStruct("Abc", types.StructData{
 		"e": types.Bool(false),
 	}), &t4, aBc{false})
+
+	// Name of struct is irrelevant to unmarshalling structs.
```
Comment (aboodman) [inline go/marshal/decode_test.go]: Can you add a test for where the struct name _does_ match and for when the destination struct is empty just for completeness?

**Comment by aboodman** [comment]
welp, that was hard.

**Comment by vinibaggio** [comment]
EDIT: Per request, added two more scenarios, one with anonymous structs and with structs that do match names.

**Comment by aboodman** [comment]
test this

**Comment by aboodman** [comment]
Jenkins: test this

**Comment by aboodman** [comment]
LGTM, just waiting for tests.

**Comment by aboodman** [comment]
💯

**Comment by arv** [comment]
LGTM

Thanks. I'll update tracking bug.
